### PR TITLE
Update .nvmrc to Node 10

### PR DESCRIPTION
**Pull request by BinaryMuse**
Node.js v12 is scheduled to me moved to LTS in a couple weeks; this moves us to the current LTS in preparation for another upgrade after.

@emplums I seem to recall you saying that this was attempted before but had some issue; do you recall what they were?

### CODE REVIEW

**Comment by emplums** [comment]
@BinaryMuse @dmarcey yeah I was having trouble running our build! But it looks like it's working fine??

**Comment by smockle** [comment]
From @dmarcey:
> @smockle did you try to do this a few weeks ago?

Indeed, I did: https://github.com/primer/components/pull/541. It was reverted—I heard it caused build issues, but I don’t know the details.

**Comment by BinaryMuse** [comment]
Okay, this is looking pretty good. The latest commit is running Node 10 (via `.nvmrc`, not sure if it's respected on GH Actions), and most dependencies have been updated:

**`npm outdated` before**

|Package|Current|Wanted|Latest|Location|
|:-|-:|-:|-:|-:|
|@babel/cli|7.5.5|7.5.5|7.6.2|@primer/components|
|@babel/core|7.5.5|7.5.5|7.6.2|@primer/components|
|@babel/plugin-proposal-object-rest-spread|7.5.5|7.5.5|7.6.2|@primer/components|
|@babel/polyfill|7.4.4|7.4.4|7.6.0|@primer/components|
|@babel/preset-env|7.5.5|7.5.5|7.6.2|@primer/components|
|@mdx-js/loader|0.15.0|0.15.0|1.5.0|@primer/components|
|@mdx-js/mdx|0.15.0|0.15.0|1.5.0|@primer/components|
|@mdx-js/tag|0.15.0|0.15.0|0.20.3|@primer/components|
|@styled-system/prop-types|5.0.5|5.0.5|5.1.2|@primer/components|
|@styled-system/theme-get|5.0.5|5.0.5|5.1.2|@primer/components|
|@svgr/webpack|2.4.1|2.4.1|4.3.3|@primer/components|
|@types/styled-system|5.1.1|5.1.1|5.1.2|@primer/components|
|babel-core|7.0.0-bridge.0|7.0.0-bridge.0|6.26.3|@primer/components|
|babel-plugin-macros|2.4.2|2.4.2|2.6.1|@primer/components|
|babel-plugin-styled-components|1.10.0|1.10.0|1.10.6|@primer/components|
|enzyme-adapter-react-16|1.5.0|1.5.0|1.14.0|@primer/components|
|eslint|4.19.1|4.19.1|6.5.1|@primer/components|
|eslint-plugin-github|1.0.0|1.0.0|3.1.3|@primer/components|
|eslint-plugin-jest|21.15.1|21.15.1|22.17.0|@primer/components|
|eslint-plugin-jsx-a11y|6.0.3|6.0.3|6.2.3|@primer/components|
|eslint-plugin-react|7.8.2|7.8.2|7.16.0|@primer/components|
|fs-extra|7.0.0|7.0.0|8.1.0|@primer/components|
|globby|8.0.1|8.0.1|10.0.1|@primer/components|
|jest|25.0.0|25.0.0|24.9.0|@primer/components|
|jest-styled-components|6.3.1|6.3.1|6.3.3|@primer/components|
|nanoid|2.0.0|2.0.0|2.1.2|@primer/components|
|node-sass|4.9.3|4.9.3|4.12.0|@primer/components|
|primer-forms|2.1.4|2.1.4|3.0.0|@primer/components|
|primer-layout|1.4.9|1.4.9|2.0.0|@primer/components|
|primer-markdown|3.7.9|3.7.9|4.0.0|@primer/components|
|primer-module-build|1.0.5|1.0.5|2.0.0|@primer/components|
|primer-navigation|1.5.7|1.5.7|2.0.0|@primer/components|
|primer-tooltips|1.5.7|1.5.7|2.0.0|@primer/components|
|primer-utilities|4.12.0|4.12.0|5.0.0|@primer/components|
|raw-loader|0.5.1|0.5.1|3.1.0|@primer/components|
|react|16.9.0|16.10.2|16.10.2|@primer/components|
|react-dom|16.9.0|16.10.2|16.10.2|@primer/components|
|react-is|16.8.6|16.8.6|16.10.2|@primer/components|
|react-router-dom|4.3.1|4.3.1|5.1.2|@primer/components|
|react-test-renderer|16.8.1|16.8.1|16.10.2|@primer/components|
|rollup|0.62.0|0.62.0|1.23.0|@primer/components|
|rollup-plugin-babel|4.0.0-beta.8|4.0.0-beta.8|4.3.3|@primer/components|
|rollup-plugin-commonjs|9.1.3|9.1.3|10.1.0|@primer/components|
|styled-components|4.1.2|4.1.2|4.4.0|@primer/components|
|styled-system|5.0.5|5.0.5|5.1.2|@primer/components|
|@loadable/component|5.10.2|5.10.3|5.10.3|docs|
|gatsby|2.15.18|2.15.29|2.15.29|docs|
|minipass|2.7.0|2.9.0|2.9.0|docs|
|react|16.9.0|16.10.2|16.10.2|docs|
|react-dom|16.9.0|16.10.2|16.10.2|docs|

**`npm outdated` after**

|Package|Current|Wanted|Latest|Location|
|:-|-:|-:|-:|-:|
|babel-core|7.0.0-bridge.0|7.0.0-bridge.0|6.26.3|@primer/components|
|jest|25.0.0|25.0.0|24.9.0|@primer/components|
|primer-forms|2.1.4|2.1.4|3.0.0|@primer/components|
|primer-layout|1.4.9|1.4.9|2.0.0|@primer/components|
|primer-markdown|3.7.9|3.7.9|4.0.0|@primer/components|
|primer-module-build|1.0.5|1.0.5|2.0.0|@primer/components|
|primer-navigation|1.5.7|1.5.7|2.0.0|@primer/components|
|primer-tooltips|1.5.7|1.5.7|2.0.0|@primer/components|
|primer-utilities|4.12.0|4.12.0|5.0.0|@primer/components|
|react-router-dom|4.3.1|4.3.1|5.1.2|@primer/components|

**Comment by BinaryMuse** [comment]
It looks like the site has some minor style changes which I'm looking into, but otherwise things seem to have gone smoothly.

**Comment by BinaryMuse** [comment]
> From @dmarcey:
> 
> > @smockle did you try to do this a few weeks ago?
> 
> Indeed, I did: #541. It was reverted—I heard it caused build issues, but I don’t know the details.

Interesting, do you know who might have more context?

**Comment by emplums** [comment]
@BinaryMuse that was me! I don't remember 100% of the details but I believe the issue was using the `sass.macro` library to build our `dist` folder

File: src/TextInput.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import React from 'react'
 import classnames from 'classnames'
 import PropTypes from 'prop-types'
+import systemPropTypes from '@styled-system/prop-types'
 import styled from 'styled-components'
+import {layout} from 'styled-system'
```
Comment (emplums) [inline src/TextInput.js]:
```suggestion
import {width} from 'styled-system'
```

File: src/TextInput.js [hunk 2]
```diff
@@ -64,6 +66,7 @@ const TextInput = styled(TextInputBase)`
     font-size: ${get('fontSizes.1')}px;
   }
   ${COMMON};
+  ${layout.width}
```
Comment (emplums) [inline src/TextInput.js]: You can import `width` directly from `styled-system`
```suggestion
  ${width}
```

File: src/__tests__/__snapshots__/LabelGroup.js.snap [hunk 3]
```diff
@@ -58,19 +56,19 @@ exports[`BranchName matches snapshot 1`] = `
...
   <span
-    className="c1"
+    className="c1 LabelItem"
     size="medium"
   >
     Default label
   </span>
   <span
-    className="c1"
+    className="c1 LabelItem"
```
Comment (emplums) [inline src/__tests__/__snapshots__/LabelGroup.js.snap]: 🤔 Why did `LabelItem` get added in this PR? I'd expect it to have been added in the 14.4.0 tracking branch but should already be on master

File: src/__tests__/__snapshots__/LabelGroup.js.snap [hunk 4]
```diff
@@ -38,8 +37,7 @@ exports[`BranchName matches snapshot 1`] = `
   padding-top: 4px;
   padding-bottom: 4px;
   background-color: #6a737d;
-  width: medium;
-  height: medium;
+  margin-right: 4px;
```
Comment (emplums) [inline src/__tests__/__snapshots__/LabelGroup.js.snap]: Curious how this margin-right got added

File: src/__tests__/__snapshots__/LabelGroup.js.snap (hunk 3 above)
Comment (emplums) [reply src/__tests__/__snapshots__/LabelGroup.js.snap]: Not too concerned about the snapshot changes though, they reflect what the accurate snapshots should look like

**Pull request merged**